sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/AccumulatedAttemptedMetricResults.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.metrics;
+
+import com.google.common.base.Function;
+
+/**
+ * Implementation of attempted {@link MetricResults} using accumulated {@link MetricsContainers}.
```

Comment:
Can you provide some more Javadoc on how a runner is expected to use this (and the other classes) to implement the metrics contract? I'm having to read the code to see how these are used to figure out how they should fit together.